---
title: "Network Analysis in Gene Expression Data using WGCNA"
author: "Alejandro Ponce Flores"
date: "`r format(Sys.Date(), '%B %d, %Y')`"
output:
  html_notebook:
    toc: true
    toc_float: true
    code_folding: "hide"
    theme: united
    highlight: tango
---

# Introduction

This notebook is dedicated to performing network analysis on gene expression data using the Weighted Gene Co-expression Network Analysis (WGCNA) package in R. The aim is to identify modules of highly correlated genes, relate these to external traits, and identify candidate biomarkers or therapeutic targets.

## Purpose

The primary objectives of this network analysis are:

1.  **Network Construction**: Building a gene co-expression network using WGCNA.
2.  **Module Detection**: Identifying modules of co-expressed genes and visualizing them.
3.  **Gene Filtering**: Refining the gene list to focus on those with significant network participation.
4.  **Correlation with Phenotype**: Linking gene modules to phenotypic traits, if available.
5.  **Eigengene Analysis**: Understanding the principal component (eigengene) of each module and its relationship to phenotypes.
6.  **Exporting Results**: Saving the modules and their associations for further analysis.

## Data Description

-   **Data Source**: [Describe the source of your expression data, e.g., public database, in-house sequencing]
-   **Expression Matrix**: Gene expression matrix derived from previous DESeq2 analysis.

## Pipeline Overview

1.  **Data Preparation for WGCNA**:
    -   Importing and transposing the expression matrix.
    -   Checking for outliers and normalizing data.

2.  **Network Construction**:
    -   Building a weighted gene co-expression network.
    -   Detecting gene modules with hierarchical clustering.

3.  **Module Visualization and Analysis**:
    -   Visualizing gene dendrograms and module colors.
    -   Analyzing module properties and gene significance.

4.  **Correlation with External Traits**:
    -   Correlating modules with phenotypic data, if available.
    -   Identifying key modules related to traits of interest.

5.  **Eigengene and Network Analysis**:
    -   Computing module eigengenes.
    -   Assessing the relationship of eigengenes with external traits.

6.  **Result Compilation and Export**:
    -   Summarizing findings and exporting data for further analysis.

### Load all required libraries

```{r message=FALSE, warning=FALSE, include=FALSE}
library(WGCNA)
library(DESeq2)
```


## Data Preparation for WGCNA

```{r}
# Load expression matrix and important genes list
expressionMatrix<- read.csv("Data/filtered_counts_data.csv")
top_genes <- read.csv("Data/top_genes_df.csv")
```

Detect Outliers and NAs
```{r}

if (any(is.na(expressionMatrix))) {
    # removing genes with any missing value
    expressionMatrix <- na.omit(expressionMatrix)
}

# Set the 'X' column as row names
row.names(expressionMatrix) <- expressionMatrix$X
expressionMatrix <- expressionMatrix[,-1]


#  outlier detection ???
```

Filter genes
```{r}

# Important genes list
all_genes <- unlist(top_genes)
all_genes <- all_genes[!is.na(all_genes)]
unique_genes <- unique(all_genes)

# Calculate variance for each gene
gene_variances <- apply(expressionMatrix, 1, var)

# Sort genes by variance while ensuring unique_genes are prioritized
sorted_genes <- names(sort(gene_variances, decreasing = TRUE))
priority_genes <- unique(c(unique_genes, sorted_genes))

# Select the top 5000 genes, ensuring no duplication
top_5000_genes <- priority_genes[1:min(5000, length(priority_genes))]

# Subset the expression matrix to include only the top 10000 genes
top_genes_matrix <- expressionMatrix[rownames(expressionMatrix) %in% top_5000_genes, ]

```


Tranpose
```{r}
# Transpose the matrix
transposed_matrix <- t(top_genes_matrix)
transposed_matrix_numeric <- apply(transposed_matrix, 2, as.numeric)
rownames(transposed_matrix_numeric) <- rownames(transposed_matrix)
```



## Network Construction

Choose the shareholding power
```{r message=FALSE, warning=FALSE}
# Choose a set of soft-thresholding powers
powers <- c(1:30)

# Call the function pickSoftThreshold to determine the best power
sft <- pickSoftThreshold(transposed_matrix_numeric, powerVector = powers, verbose = 5)

```


Plot connectivity
```{r fig.height=4, fig.width=12, warning=FALSE}
# Open a plotting window
par(mfrow=c(1,2))

# Open a plotting window with three panels (1 row, 3 columns)
par(mfrow=c(1,3))

# Plot 1: Scale Independence (Scale-Free Topology Fit, R²)
plot(sft$fitIndices$Power, sft$fitIndices$SFT.R.sq, 
     xlab="Soft Threshold (power)", ylab="Scale Independence (R^2)", 
     type="b", main="Scale Independence",
     col="blue")
abline(h=0.8, col="red", lty=2)  # Reference line

# Plot 2: Mean Connectivity
plot(sft$fitIndices$Power, sft$fitIndices$mean.k, 
     xlab="Soft Threshold (power)", ylab="Mean Connectivity", 
     type="b", main="Mean Connectivity",
     col="green")

# Plot 3: Scale-Free Topology Fit Index (R²)
plot(sft$fitIndices$Power, -log10(sft$fitIndices$truncated.R.sq), 
     xlab="Soft Threshold (power)", ylab="-log10 Scale Free Topology Model Fit (R^2)",
     type="b", main="Scale-Free Topology Fit",
     col="orange")
```




```{r}
# Network construction
net <- blockwiseModules(transposed_matrix_numeric, 
                        power = 5,            # Soft-thresholding power; choose based on scale-free fit analysis (usually between 6-14)
                        TOMType = "signed",   # Type of Topological Overlap Matrix. Use "unsigned" if you don't want to consider the sign of correlations
                        minModuleSize = 30,   # Minimum number of genes in a module; adjust based on your dataset size and complexity
                        reassignThreshold = 10, # Threshold for reassigning genes to modules, higher value means more aggressive reassignment
                        mergeCutHeight = 0.1, # Height for merging modules; adjust this to control the degree of module merging (0-1)
                        numericLabels = TRUE, # If TRUE, labels modules numerically; if FALSE, labels them by colors
                        pamRespectsDendro = FALSE, # If TRUE, PAM stage respects the dendrogram; if FALSE, it does not. Depends on your data structure
                        saveTOMs = TRUE,       # If TRUE, saves the TOMs for later use; useful for large datasets and repeated analyses
                        saveTOMFileBase = "TOM", # Base name for saved TOM files
                        verbose = 1,            # Controls how much information to print (0 = no output, higher numbers = more output)
                        ds = 3)                 # Network construction algorithm; use 1 or 2 for large datasets, 3 for smaller datasets

```


## Module Visualization and Analysis

```{r}
# Convert module labels to colors for visualization
mergedColors = labels2colors(net$colors)

# Plot the dendrogram and module colors
plotDendroAndColors(net$dendrograms[[1]], mergedColors[net$blockGenes[[1]]],
                    "Module colors",
                    dendroLabels = FALSE, hang = 0.03,
                    addGuide = TRUE, guideHang = 0.05)

```

## Correlation with External Traits

Save module information
```{r}
# Extract gene names from the transposed matrix
genes = colnames(transposed_matrix)
# Convert module labels to colors
moduleColors = labels2colors(net$colors)
# Combine gene names with their module colors
mymodules = cbind(genes, moduleColors)
# Save the data to a file
save(mymodules, file = "WGCNA_modules.RData")
```

Create phenotype vector and module Eigenes - Tumor/Normal
```{r}
# Extract sample names
sampleNames = rownames(transposed_matrix_numeric)

# Create a phenotype vector
phenotype = ifelse(grepl("\\.N$", sampleNames), 0, 1)

# Compute module eigengenes
MEs0 = moduleEigengenes(transposed_matrix_numeric, colors = net$colors)$eigengenes

# Correlate module eigengenes with phenotype
moduleTraitCor = cor(MEs0, phenotype, use = "p")

# p-values for the correlations
moduleTraitPvalue = corPvalueStudent(moduleTraitCor, nrow(transposed_matrix_numeric))

```

!!!
If you continue with this data set and want to do something fun with WGCNA (absolutely not required - just a suggestion!) you can use it to draw a consensus network with control and normal and compare the two.!!!!
